src/components/features/dragons.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { getDragonData, reserveD, cancelD } from '../../redux/dragons/dragonsSlice';
+import '../styles/dragonStyle.css';
+
+function Dragons() {
+  const dragons = useSelector((state) => state.dragons);
+  const loading = useSelector((state) => state.dragons.loading);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getDragonData());
+  }, [dispatch]);
+
+  const handleReserveBtn = (id) => {
+    dispatch(reserveD(id));
+  };
+  const handleCancelBtn = (id) => {
+    dispatch(cancelD(id));
+  };
+
+  if (loading) {
+    return <div>loading....</div>;
+  }
```

Comment:
- [ ] Well done :clap:   you guys did a very nice job :+1: but there are some problems with your dragon page when we click on the `reserve button` it changes to the `cancel button` it's good :+1: but when we go to the next page and come back to the dragon page the state is empty so kindly fix this issue :+1: check below :point_down: my explanation video then you will understand better :slightly_smiling_face: 

[screen-capture (1).webm](https://user-images.githubusercontent.com/45226848/236560980-56360f02-9a39-47ed-93f7-89ea4de1189b.webm)